test/powershell/engine/Module/ModuleCmdlet.Tests.ps1

```diff
@@ -0,0 +1,518 @@
+
+    Context "Parameter handling" {
+        BeforeAll {
+            SetupModules
+
+            $validationTests = @(
+                @{ paramName="Variable" },
+                @{ paramName="Function" },
+                @{ paramName="Cmdlet" },
+                @{ paramName="Variable" }
+            )
+
+            $simpleModContent = @'
+function Test-FirstModuleFunction
+{
+    Write-Output "TESTSTRING"
+}
+'@
+
+            $ps = [powershell]::Create()
+        }
+
+        AfterAll {
+            TearDownModules
+            $ps.Dispose()
+        }
+
+        AfterEach {
+            $ps.Streams.ClearStreams()
+        }
+
+        It "Validates a null -<paramName> parameter" -TestCases $validationTests {
+            param($paramName)
+
+            $modData = New-TestModule $simpleModContent
+            $sb = [scriptblock]::Create("Import-Module $($modData.Path) -$paramName `$null")
+            $sb | Should -Throw -ErrorId "ParameterArgumentValidationError"
+        }
+
+        It "Ignores whitespace -MaximumVersion parameter" {
+            $modData = New-TestModule $simpleModContent
+            Import-Module $modData.Path -MaximumVersion "    "
+            Test-FirstModuleFunction | Should -BeExactly "TESTSTRING"
+        }
+    }
+}
+
+Describe "Import-Module by PSModuleInfo" -Tag "Feature" {
+    BeforeAll {
+        SetupModules
+
+        $simpleModContent = @'
+function Test-FirstModuleFunction
+{
+    Write-Output "TESTSTRING"
+}
+'@
+    }
+
+    AfterAll {
+        TearDownModules
+    }
+
+    BeforeEach {
+        $modData = New-TestModule -Content $simpleModContent
+        $modInfo = Import-Module $modData.Path -PassThru
+        Remove-Module $modInfo
+    }
+
+    It "Imports a module by moduleinfo" {
+        { Test-FirstModuleFunction } | Should -Throw -ErrorId "CommandNotFoundException"
+
+        Import-Module $modInfo
+
+        Test-FirstModuleFunction | Should -BeExactly "TESTSTRING"
+    }
+
+    It "Passes the imported module out when -PassThru is used" {
+        $mod = Import-Module $modInfo -PassThru
+
+        $mod.ExportedFunctions.Keys | Should -Contain "Test-FirstModuleFunction"
+    }
+
+    It "Passes the imported module out as a custom object when -AsCustomObject is used" {
+        $mod = Import-Module $modInfo -AsCustomObject
+
+        $mod                              | Should -BeOfType "PSCustomObject"
+        $mod.'Test-FirstModuleFunction'() | Should -BeExactly "TESTSTRING"
+    }
+
+    It "Re-imports modules from file with new members when -Force is used" {
+        $module = Import-Module $modInfo -PassThru
+
+        $module.ExportedFunctions.Count | Should -Be 1
+        $module.ExportedFunctions.Keys  | Should -Contain "Test-FirstModuleFunction"
+        $module.ExportedFunctions.Keys  | Should -Not -Contain "Test-SecondModuleFunction"
+
+        $newModuleContent = (Get-Content $modData.Path | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }"
```

Comment:
> Get-Content $modData.Path | Out-String

Why not use `$simpleModContent`? Or having `$newModuleContent` declared in `beforeall` like `$simpleModContent`?